### DISCUSSION TITLE
Multiple table field rows from CSV

### DISCUSSION THREAD

**Discussion by colinmeinke** [engram-design/FeedMe#881] [Ideas]
### Description

The mapping from a CSV column to a table field column works great, but how can you add more than one row to a table? I was hoping that an entry could spread over more than one row in the CSV, like this https://file-ewgbcvrimc.now.sh. But this doesn't work, as feedme expects each line to be an entry.

### Additional info

- Plugin version: latest version of pro
- Craft version: 3

**Comment by engram-design** [comment]
Complex fields like Matrix or Table aren't really supported in the CSV format. I'm also hesitant to ever add support for them due to unexpected results. In your example, its pretty ambiguous as to what data should be applicable with a single entry. Just because there's no values on the first few columns, sounds mean its part of the previous row's data. Maybe this new row should be data for a new entry?

If anything, this will likely take the form or piping values, like elements work (Category 1|Category 2, etc). Or even using JSON-encoded data, but that really seems like we're using the wrong data type.

**Comment by colinmeinke** [comment]
JSON would be ideal, but I'm working with a client that needs to be able to create and import this data themselves. CSV is a good fit for this.

I guess an option could be to mark a column as required, and if that doesn't exist for a row, treat the row as supplementary data to the previous entry?

As a second best option, piping would also work, but this wouldn't be as clear from the CSV perspective.

**Comment by kdouglass** [comment]
Having the option to add rows with piping would be ideal for CSVs, in a simple use case. I'm looking for a way to basically create a simple textfield repeater and a table seems to be the best approach for CSVs and Feed Me. 

Anyway to get this implemented?

**Comment by engram-design** [comment]
At the moment, piping is supported to create multiple rows. For example:

```csv
Entry Title,Column 1,Column 2
New Blog Entry,Row 1 Value|Row 2 Value,Row 1 Value|Row 2 Value
```

The above will result the the content for the Table field to be:
Row 1 Value    Row 1 Value
Row 2 Value    Row 2 Value

**Reply by knynkwl** [reply]
I am currently trying to implement this and it doesn't seem to be working, using 4.4.0v, any ideas?
<img width="410" alt="Screen Shot 2022-10-24 at 10 50 36 AM" src="https://user-images.githubusercontent.com/4225771/197556147-7d029431-1ed8-4948-9757-14d8f63626dc.png">
